usr/share/rear/restore/SUSE_LINUX/91_create_missing_directories.sh

```diff
@@ -1,7 +1,23 @@
 #
-# SuSE likes to have stuff under /media
-#
-# create missing directories
-pushd /mnt/local >&8
-mkdir -p media/cdrom media/floppy
+# In some cases SUSE has /media plus /media/cdrom and /media/floppy.
+# On SLE11 the filesystem RPM contains them.
+# On SLE12 they are dropped from the filesystem RPM because nowadays /run/media is used.
+# There is the SUSE-internal issue https://bugzilla.suse.com/show_bug.cgi?id=890198
+# and the "SUSE Linux Enterprise Server 12 Release Notes" mention it
+# see https://www.suse.com/releasenotes/x86_64/SUSE-SLES/12/ that reads:
+#   "/run/media/<user_name> is now used as top directory for removable
+#    media mount points. It replaces /media , which is not longer available."
+# Therefore for SLE12 "rear recover" must no longer create them.
+# FIXME: The following test for SLE12 products is intentionally sloppy
+# because I <jsmeix@suse.de> have no better idea how to test for various
+# posible SLE12-based products like "SUSE Linux Enterprise Server 12"
+# "SUSE Linux Enterprise Desktop 12" "SUSE Linux Enterprise Server 12 SP1"
+# "SUSE Linux Enterprise Desktop 12 SP1" "SUSE Linux Enterprise <whatever> <whichever>".
```

Comment:
"((  OS_VERSION < 12 ))" was the first thing I tried but
it cannot work because OS_VERSION can be things
like "13.2" for openSUSE 13.2 and then:
<pre>
# echo $OS_VERSION
13.2

# if ((  OS_VERSION < 12 )) ; then echo y ; fi
-bash: ((: 13.2: syntax error: invalid arithmetic operator (error token is ".2")
</pre>

Perhaps an OS_VERSION setting of "13.2" is wrong
and needs to be fixed?
But at least currently such an OS_VERSION is allowed
and also used for Red Hat according to
doc/user-guide/02-getting-started.adoc
<pre>
# rear dump
Relax-and-Recover 1.12.0svn497 / 2011-07-11
Dumping out configuration and system information
System definition:
                                    ARCH = Linux-x86_64
                                      OS = GNU/Linux
                               OS_VENDOR = RedHatEnterpriseServer
                              OS_VERSION = 5.6
</pre>

But even if OS_VERSION would be split into
<pre>
OS_MAJOR_VERSION=13
OS_MINOR_VERSION=2
</pre>
it does not really help me in this particular case because
I simply do not know for what exact openSUSE versions
media/cdrom and media/floppy were used and I do not want
to install each openSUSE version to find out what there
actually is.

As I said:
In general I do not like testing for version numbers.
In general I prefer testing for the real thing.

In this particular case I implemented "testing for the real thing"
in a sloppy and indirect way (I know RFC 1925 6a) but
personally I consider this to be still better than obscure
version number tests and I provide intentionally a longer
explanatory comment to tell the user what the issue is.

What does a plain number tell the user?
Basically nothing.

Finally the whole
usr/share/rear/restore/SUSE_LINUX/91_create_missing_directories.sh
looks broken to me on a first glance because I wonder why
directories in /media are not in the backup?

As far as I see on my SLE11 system /media is not a separated filesystem so that I would expect to have all in /media included
in the backup by default.

In contrast on my SLE12 system /run is a separeted filesystem:
<pre>
# findmnt | grep run  
|-/run  tmpfs  tmpfs  rw,nosuid,nodev,mode=755
</pre>
so that things in /run/media should not be in the backup
by default.